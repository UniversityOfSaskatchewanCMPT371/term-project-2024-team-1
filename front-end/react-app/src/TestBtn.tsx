import React from 'react';

interface ButtonProps {
  label: string;
  onClick: () => void;
}

const Button: React.FC<ButtonProps> = ({ label, onClick }) => (
  <button id="mockButton" onClick={onClick}>{label}</button>
);

export default Button;