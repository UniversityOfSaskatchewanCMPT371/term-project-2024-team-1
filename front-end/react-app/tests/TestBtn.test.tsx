
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TestBtn from '../src/TestBtn'


describe('Button Component', () => {
  it('should match the snapshot', () => {
    const { asFragment, getByText } = render(
      <TestBtn label="Click me" onClick={() => console.log('Button clicked')} />
    );

    // Assert that the rendered component matches the snapshot
    expect(asFragment()).toMatchSnapshot();

  

    // Optionally, you can simulate user interaction and test the resulting changes
    fireEvent.click(getByText('Click me'));

    // After user interaction, assert that the updated component matches the snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});