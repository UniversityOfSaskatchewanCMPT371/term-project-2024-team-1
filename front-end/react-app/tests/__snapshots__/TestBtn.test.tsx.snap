// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match the snapshot 1`] = `
<DocumentFragment>
  <button
    id="mockButton"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button Component should match the snapshot 2`] = `
<DocumentFragment>
  <button
    id="mockButton"
  >
    Click me
  </button>
</DocumentFragment>
`;
